classes/local/course_enrolment_manager.php

```diff
@@ -53,10 +55,17 @@ public function search_users_with_groups(string $search = '', bool $searchanywhe
                  JOIN {enrol} e ON ue.enrolid = e.id
                  JOIN ({groups_members} gm JOIN {groups} g ON (g.id = gm.groupid))
                       ON (u.id = gm.userid AND g.courseid = e.courseid)
+                 JOIN {role_assignments} ra ON ra.userid = u.id
                 WHERE $wherecondition
                   AND e.courseid = :courseid
+                  AND u.suspended = 0
+                  AND 1 = (SELECT permission FROM {role_capabilities}
```

Comment:
this doesn't look right to me? - 1 = (SELECT permission) ? won't this always fail?

Reply:
@danmarsden 
See below from the DB. The roleid 5 (student) row is what will be feeding to the above query. 

```
select * from mdl_role_capabilities where capability like 'mod/dialogue:receive';
  id  | contextid | roleid |      capability      | permission | timemodified | modifierid 
------+-----------+--------+----------------------+------------+--------------+------------
 1525 |         1 |      5 | mod/dialogue:receive |          1 |   1738014397 |          0
 1526 |         1 |      4 | mod/dialogue:receive |          1 |   1738014397 |          0
 1527 |         1 |      3 | mod/dialogue:receive |          1 |   1738014397 |          0
 1528 |         1 |      1 | mod/dialogue:receive |         -1 |   1738014397 |          0
111